File: 443-string-compression/string-compression.java
class Solution {
    public int compress(char[] a) {
        int index = 0;
        int count = 0;

        for (int i = 0; i < a.length; i++) {
            count++;
            if (i == a.length - 1 || a[i] != a[i + 1]) {
                a[index] = a[i];

                if (count > 1) {
                    index++;

                    if (count > 9) {
                        String str = Integer.toString(count);
                        for (char c : str.toCharArray()) {
                            a[index] = c;
                            index++;
                        }
                    } else {
                        a[index] = (char)(count + '0');
                        index++;
                    }
                } else {
                    index++;
                }
                count = 0;
            }
        }
        return index;
    }
}

File: 443-string-compression/string-compression.py
class Solution:
    def compress(self, c: List[str]) -> int:
        n = len(c)
        count = 1
        s = ""
        i = 0
        for i in range(1, n):
            if c[i] != c[i - 1]:
                if count == 1:
                    s = s + c[i - 1]
                else:
                    s = s + c[i - 1] + str(count)
                count = 1
            else:
                count += 1
        
        
        if count == 1:
            s = s + c[i]
        else:
            s = s + c[i] + str(count)

        for i in range(len(s)):
            c[i] = s[i]

        return len(s)
